### DISCUSSION TITLE
21.09.12 일 회의록

### DISCUSSION THREAD

**Discussion by damilog** [Team-IT-DA/Frontend#143] [회의록]
## 논의 사항
- [x] orderList mock data 제니가 정한건지 아님 api 기준인지?? (데이지) - 임의로 하신 것 
- [x] orderList props 늘리더라도 atom으로 만들기 vs OrderList atom + 추가 UI 한 OrderSheet별도 생성하기 - orderList 용 컴포넌트 하나 만들기

## 공유 사항 
- myPage 폴더 내 각 탭 page 별로 폴더 만들어서 관리하기
- Recoil Selector로 유저 검증하는걸 계속 요청해서 상태로 가지고 있기?
- 만약 로컬스토리지에 관리자인지 아닌지 정보를 담아놓아서 뷰를 조작한다해도 유효성 검증을 서버, 클라이언트 둘 다 할 거니..크게 문제는 안 될듯..하지만 고민 필요

### Daisy
- [x] OrderList 분리해서 OrderCard 만들기
- [x] Cart 페이지 모달 토글 기능 완료
- [x] 각자 만든거 살펴보면서 리팩토링

### Dico
- [x] input Atom 적용하기 
- [x] myPage 등록 상품 조회 탭 만들기
- [x] 개인 정보 수정에 판매자용 하위 탭 만들기 => 1안으로 해보고 안되겠다 싶으면 2안으로 가기
- [ ] 각자 만든거 살펴보면서 리팩토링

### Jenny
- [x] 에디터 영역 로직 살펴보기 (API 읽어보기)
- [x] 각자 만든거 살펴보면서 리팩토링

**Comment by wjddnjswjd12** [comment]
백엔드한테 말해둬야 할 점:
OrderList의 각 Order의 product마다 추가 가능한지
- `hasReview` 라는 필드 추가 가능한지
- 사용자가 로컬에서 파일첨부를 눌렀을 때 Image파일을 어디로 보내야 하는지? 어떻게 처리해야 하는지
- MyPage > 판매자 페이지 API 설계 필요

**Comment by damilog** [comment]
- https://kosaf04pyh.tistory.com/152
- https://maxkim-j.github.io/posts/react-query-preview
- 캐싱을 할 때 캐싱을 하고 refetch 할 때 refetch 함

**Comment by ha3158987** [comment]
< 개략적으로 알아본 사용자 유형을 저장하는 흐름 (이 외에도 다양한 방법이 있는 듯 합니다) > 
1. 사용자가 로그인을 하면 서버에서 발급해주는 access token을 받는다. 
2. 브라우저 cookie에 access token을 저장한다. => withCredentials
3. 페이지가 바뀌면 DOM을 그리기 '전'에 가지고 있는 access token이 있는 지 확인한다.  있다면 get요청 시 백엔드에 access token을 함께 전달해서 유효성 검사를 한다. 
4. 유효성 검사에서 사용자 유형(판매자 or 구매자) 검증이 되면 그 정보를 바탕으로 개별화된 페이지/컴포넌트를 보여준다. 
5. 로그인이 필요한 매 페이지마다 axios, react query, useSWR 등으로 fetch를 요청한다.
